### ISSUE TITLE
Regression: terraform cannot create google_computer_instance_group_manager in other region than project

### ISSUE THREAD

**Issue by bastiaanb** [hashicorp/terraform#9344]
### Terraform Version

0.7.1-0.7.5: tested with 0.7.0, 0.7.1, 0.7.2 and 0.7.5. Version 0.7.0 is OK, 0.7.1 and higher show regression.
### Affected Resource(s)
- ~~google_compute_autoscaler~~
- google_computer_instance_group_manager

The issue is with the google_computer_instance_group_manager, not the autoscaler that tries to target it. See comment below
### Terraform Configuration Files

``` hcl
provider "google" {}

variable "zones" {
  default = ["europe-west1-d", "us-east1-b"]
}

  resource "google_compute_autoscaler" "nomad_client_scaler" {
  count = 2
  name = "${format("nomad-client-scaler-%d", count.index)}"
  zone = "${element(var.zones, count.index)}"

  target = "${element(google_compute_instance_group_manager.nomad_client_group.*.self_link, count.index)}"
  autoscaling_policy = {
    min_replicas = 1
    max_replicas = 2
    cooldown_period = 60
    cpu_utilization = {
      target = 0.75
    }
  }
}

resource "google_compute_instance_group_manager" "nomad_client_group" {
  count = 2
  name = "${format("nomad-client-group-%d", count.index)}"
  zone = "${element(var.zones, count.index)}"

  description = "Group consisting of Nomad client nodes"
  instance_template = "${google_compute_instance_template.nomad_client.self_link}"
  base_instance_name = "${format("client-%d", count.index)}"
}

resource "google_compute_instance_template" "nomad_client" {
  name = "client"
  description = "Template for Nomad client nodes"
  instance_description = "Nomad client node"
  machine_type = "f1-micro"
  tags = ["nomad", "client"]

  scheduling {
    automatic_restart = true
    on_host_maintenance = "MIGRATE"
    preemptible = false
  }

  disk {
    source_image = "coreos-stable-1122-2-0-v20160906"
    auto_delete = true
    boot = true
  }

  network_interface {
    network = "default"
    access_config {}
  }

  metadata {
  }
}
```
### Debug Output

https://gist.github.com/bastiaanb/864db1bceb0682f5efd83498b44bb1d2
Contains stderr and stdout of 0.7.5 and stdout of 0.7.0 for comparison.
### Panic Output

none
### Expected Behavior

google_compute_autoscalers should have been created, with autoscaler #0 pointing to instance group #0 and autoscaler #1 to instance group #1, like done by version 0.7.0.
### Actual Behavior

Both autoscaler instances have the first instance group as target, which coincidentally fails due to different zones.
### Steps to Reproduce
1. `terraform apply`
### Important Factoids

No 'atypical' settings or accounts.
### References

None

**Comment by bastiaanb** [comment]
Debug output shows that one instance group manager is 'removed': 
```
2016/10/12 23:06:32 [DEBUG] plugin: terraform: google-provider (internal) 2016/10/12 23:06:32 [WARN] Removing Instance Group Manager "nomad-client-group-1" because it's gone
```
This would explain why both autoscalers get the same instance group manager: the list contains only the first group, so the element() function wraps.

It looks like the resourceComputeInstanceGroupManagerRead() function assumes that all instance group managers are created in the region of the project, see 

```
	region, err := getRegion(d, config)
...
	resource, err := getZonalResourceFromRegion(getInstanceGroupManager, region, config.clientCompute, project)
```

Introduced in commit https://github.com/hashicorp/terraform/commit/da1cf0d83620baf1ca2d07069fc32377ebd6fc0a

This assumption is wrong, as GCP does not limit projects to a single region and breaks for example global load balancer setups.

**Comment by zbjornson** [comment]
I think this is the same as #9126 (where it's not marked as a regression).

**Comment by stack72** [comment]
Closed via #11294
